### PR TITLE
Merge cawolf

### PR DESCRIPTION
**Pull request by 8p8c**
Hello again Christian. I've fixed bug with calculating steps to show in Scenario Outline with Background. And fixed failing steps.

### CODE REVIEW

**Comment by 8p8c** [comment]
According to this page:
https://relishapp.com/cucumber/cucumber/docs/formatters/json-output-formatter
Scenario should contain 'description' even if it is empty.

File: features/eat-cukes.feature [hunk 1]
```diff
@@ -24,14 +24,12 @@ Feature: Eat cukes Example
...
           "id": "eat-cukes-in-lot",
-          "description": "",
           "name": "Eat cukes in lot",
           "keyword": "Feature",
           "line": 1,
           "elements": [
             {
-              "id": "eat-cukes-in-lot;eating-many-cukes---11",
-              "description": "",
+              "id": "eat-cukes-in-lot;eating-many-cukes",
```
Comment (cawolf) [inline features/eat-cukes.feature:32]: I think this actually will be a problem for tools like https://wiki.jenkins.io/display/JENKINS/Cucumber+Living+Documentation+Plugin , as the `id` is not unique anymore. I will check this with one of my projects and let you know about it.

File: src/Formatter/Formatter.php [hunk 2]
```diff
@@ -225,7 +266,7 @@ public function onBeforeOutlineTested(BehatEvent\BeforeOutlineTested $event)
         $scenario->setName($event->getOutline()->getTitle());
         $scenario->setTags($event->getOutline()->getTags());
         $scenario->setLine($event->getOutline()->getLine());
-        $scenario->setType('scenario_outline');
+        $scenario->setType('scenario');
```
Comment (cawolf) [inline src/Formatter/Formatter.php:269]: I will also check this to be compatible with the Jenkins plugin I mentioned before.

**Review by cawolf — COMMENTED** [review]
As soon as I checked the compatibility, I will let you know.

File: features/eat-cukes.feature (hunk 1 above)
Comment (8p8c) [reply features/eat-cukes.feature:32]: Agree. But I can't see where I could break it. Seems I didn't touched anything concerning scenario ID.

File: src/Renderer/JsonRenderer.php [hunk 3]
```diff
@@ -107,9 +110,9 @@ protected function processScenario(Node\Scenario $scenario)
             'keyword' => $scenario->getKeyword(),
             'name' => $scenario->getName(),
             'line' => $scenario->getLine(),
-            'description' => $scenario->getDescription(),
             'type' => $scenario->getType(),
             'steps' => [],
+            'description' => $scenario->getDescription(),
```
Comment (8p8c) [inline src/Renderer/JsonRenderer.php]: Oops. Looks I'd better restore your original line in original place.

File: features/eat-cukes.feature (hunk 1 above)
Comment (cawolf) [reply features/eat-cukes.feature:32]: I'm sorry, but this breaks another plugin we are using. The `id` must be completely unique within the result file.

In the current state, I can not merge this without causing much effort.

File: src/Formatter/Formatter.php (hunk 2 above)
Comment (cawolf) [reply src/Formatter/Formatter.php:269]: See above

**Comment by cawolf** [comment]
As there are major breaking changes and no way to work around them, I can currently not accept this MR, sorry.

**Pull request closed without merging**